### PR TITLE
Don't generate line mappings for imports

### PR DESCRIPTION
**Pull request by ajaybhargavb**
Issue - #1079 

@NTaylorMullen @rynowak

### CODE REVIEW

File: src/Microsoft.AspNetCore.Razor.Evolution/CodeGeneration/DesignTimeCSharpRenderer.cs [hunk 1]
```diff
@@ -287,21 +287,6 @@ public override void VisitDeclareTagHelperFields(DeclareTagHelperFieldsIRNode no
             }
         }
 
-        private void AddLineMappingFor(RazorIRNode node)
```
Comment (ajaybhargavb) [inline src/Microsoft.AspNetCore.Razor.Evolution/CodeGeneration/DesignTimeCSharpRenderer.cs]: This was a duplicate `AddLineMappingFor` method.

File: src/Microsoft.AspNetCore.Razor.Evolution/CodeGeneration/CSharpRenderingContext.cs [hunk 2]
```diff
@@ -39,7 +39,9 @@ internal CSharpRenderingConventions RenderingConventions
 
         internal IList<RazorDiagnostic> Diagnostics { get; } = new List<RazorDiagnostic>();
 
-        internal RazorSourceDocument SourceDocument { get; set; }
+        internal RazorCodeDocument CodeDocument { get; set; }
+
+        internal RazorSourceDocument SourceDocument { get => CodeDocument.Source; }
```
Comment (NTaylorMullen) [inline src/Microsoft.AspNetCore.Razor.Evolution/CodeGeneration/CSharpRenderingContext.cs]: `internal RazorSourceDocument SourceDocument { get => CodeDocument.Source; }` can be written as `internal RazorSourceDocument SourceDocument => CodeDocument.Source;`

Comment (ajaybhargavb) [inline src/Microsoft.AspNetCore.Razor.Evolution/CodeGeneration/CSharpRenderingContext.cs]: Just realized this should be `CodeDocument?.Source`

File: src/Microsoft.AspNetCore.Razor.Evolution/CodeGeneration/CSharpRenderingContext.cs [hunk 3]
```diff
@@ -60,6 +62,19 @@ public void AddLineMappingFor(RazorIRNode node)
                 return;
             }
 
+            var imports = CodeDocument.GetImportSyntaxTrees();
+            if (imports != null)
+            {
+                foreach (var import in imports)
+                {
+                    if (string.Equals(import.Source.FileName, node.Source.Value.FilePath, StringComparison.Ordinal))
```
Comment (NTaylorMullen) [inline src/Microsoft.AspNetCore.Razor.Evolution/CodeGeneration/CSharpRenderingContext.cs]: Case insensitive

File: src/Microsoft.AspNetCore.Razor.Evolution/CodeGeneration/CSharpRenderingContext.cs [hunk 4]
```diff
@@ -60,6 +62,19 @@ public void AddLineMappingFor(RazorIRNode node)
                 return;
             }
 
+            var imports = CodeDocument.GetImportSyntaxTrees();
+            if (imports != null)
+            {
+                foreach (var import in imports)
```
Comment (NTaylorMullen) [inline src/Microsoft.AspNetCore.Razor.Evolution/CodeGeneration/CSharpRenderingContext.cs]: if imports is a list iterate for a `for` to not allocate.

File: test/Microsoft.AspNetCore.Razor.Evolution.Test/TestFiles/IntegrationTests/CodeGenerationIntegrationTest/BasicImports_Imports0.cshtml [hunk 5]
```diff
@@ -2,3 +2,4 @@
 @using System.Globalization
 @("And also this")
 @using System.ComponentModel
+@addTagHelper "SomeType, SomeNamespace"
```
Comment (NTaylorMullen) [inline test/Microsoft.AspNetCore.Razor.Evolution.Test/TestFiles/IntegrationTests/CodeGenerationIntegrationTest/BasicImports_Imports0.cshtml]: Use inherits instead to verify this.

**Comment by ajaybhargavb** [comment]
🆙 📅

File: src/Microsoft.AspNetCore.Razor.Evolution/CodeGeneration/CSharpRenderingContext.cs [hunk 6]
```diff
@@ -60,6 +62,19 @@ public void AddLineMappingFor(RazorIRNode node)
                 return;
             }
 
+            var imports = CodeDocument.GetImportSyntaxTrees();
+            if (imports != null)
+            {
+                for (var i = 0; i < imports.Count; i++)
+                {
+                    if (string.Equals(imports[i].Source.FileName, node.Source.Value.FilePath, StringComparison.OrdinalIgnoreCase))
```
Comment (rynowak) [inline src/Microsoft.AspNetCore.Razor.Evolution/CodeGeneration/CSharpRenderingContext.cs]: This is actually doing it the hard way. You already have the source document, and the source document has a file name. You just want to match on `!=` on the source document.

Comment (ajaybhargavb) [inline src/Microsoft.AspNetCore.Razor.Evolution/CodeGeneration/CSharpRenderingContext.cs]: Good point 👍. I'll add this change in another PR.

Comment (ajaybhargavb) [inline src/Microsoft.AspNetCore.Razor.Evolution/CodeGeneration/CSharpRenderingContext.cs]: Do we even need the `CodeDocument` property anymore? IMO it's nice to have but I am okay either way.

Comment (rynowak) [inline src/Microsoft.AspNetCore.Razor.Evolution/CodeGeneration/CSharpRenderingContext.cs]: I dunno, we haven't 'cleaned' this code yet, so I guess either one you want is fine.

Comment (ajaybhargavb) [inline src/Microsoft.AspNetCore.Razor.Evolution/CodeGeneration/CSharpRenderingContext.cs]: Okay. I'll leave it here. We'll revisit during the cleanup.

Comment (ajaybhargavb) [inline src/Microsoft.AspNetCore.Razor.Evolution/CodeGeneration/CSharpRenderingContext.cs]: https://github.com/aspnet/Razor/pull/1116

**Pull request merged**